### DISCUSSION TITLE
Method for placing a player on a specific seat in a vehicle

### DISCUSSION THREAD

**Discussion by TimofeyTanuki** [SmartlyDressedGames/Unturned-3.x-Community#3702] [Ideas]
Why not add a method for putting a player on a certain seat in a car? I had the need to force the player to sit in the back seat in the vehicle, but I have no method for that. There is `VehicleManager.ServerForcePassengerIntoVehicle(SteamPlayer.player, InteractableVehicle)`, but it sits on the first matching seat.

**Comment by educatalan02** [comment]
> Why not add a method for putting a player on a certain seat in a car? I had the need to force the player to sit in the back seat in the vehicle, but I have no method for that. There is `VehicleManager.ServerForcePassengerIntoVehicle(SteamPlayer.player, InteractableVehicle)`, but it sits on the first matching seat.

InteractableVehicle.swapPlayer

**Reply by TimofeyTanuki** [reply]
I mean to put a player immediately when added to a certain seat. Also, I tried to use this method, but it didn't work. I'm not sure, but it probably only works on the server side, if it works at all.

**Reply by rube200** [reply]
Almost every call in `InteractableVehicle` is sided that means it doesnt sent to client/server, the class responsible for that is `VehicleManager`.

What you want is these: [VehicleManager.ReceiveSwapVehicleRequest](https://github.com/Unturned-Datamining/Unturned-Datamining/blob/4c054af8ffcd0c1eeddb8053dd51ffe498da9274/Assembly-CSharp/SDG.Unturned/VehicleManager.cs#L1415) asl you can try call this RPC [VehicleManager.SendEnterVehicle](https://github.com/Unturned-Datamining/Unturned-Datamining/blob/4c054af8ffcd0c1eeddb8053dd51ffe498da9274/Assembly-CSharp/SDG.Unturned/VehicleManager.cs#L88) but you need to do some check like [VehicleManager.ReceiveEnterVehicleRequest](https://github.com/Unturned-Datamining/Unturned-Datamining/blob/4c054af8ffcd0c1eeddb8053dd51ffe498da9274/Assembly-CSharp/SDG.Unturned/VehicleManager.cs#L1224).

**Reply by TimofeyTanuki** [reply]
why not just add a seat index argument for this method `VehicleManager.ServerForcePassengerIntoVehicle(SteamPlayer.player, InteractableVehicle)`?
